File: 剑指Offer/10.青蛙跳台阶问题.py
class Solution:
    def numWays(self, n: int) -> int:
        if n < 2:
            return 1

        oneStep = 1
        twoStep = 1
        for i in range(2, n + 1):
            temp = twoStep
            twoStep = oneStep
            oneStep = temp + twoStep
            oneStep %= int(1e9 + 7)
        return oneStep

File: 剑指Offer/10.青蛙跳台阶问题.cpp
class Solution {
public:
    int numWays(int n) {
        if (n < 2) return 1;
    
        int index = 2;
        int oneStep = 1;
        int twoStep = 0;
        int temp;
        while (index <= n) {
            temp = twoStep;
            twoStep = oneStep;
            oneStep = temp + twoStep;
            oneStep %= 1e9 + 7;
        }
        return oneStep;
    }
};